File: Python/002_Add_Two_Numbers.py
class Solution(object):
    def addTwoNumbers(self, l1, l2): 
        temp = 0 
        if l1 == None :
            return l2
        if l2 == None : 
            return l1
        res = l1 
        res.val += l2.val 
        if res.val >= 10 : 
            res.val -= 10 
            temp = 1 
        start = res 
        l1 = l1.next 
        l2 = l2.next 
        while l1 and l2 : 
            res.next = l1
            res = res.next 
            res.val += l2.val
            res.val += temp  
            temp = 0 
            if res.val >= 10 : 
                temp = 1 
                res.val -= 10 
            l1 = l1.next
            l2 = l2.next 
        while l1 : 
            res.next = l1 
            res = res.next 
            res.val += temp 
            temp = 0 
            if res.val >= 10 : 
                res.val -= 10 
                temp = 1 
            l1 = l1.next 
        while l2 : 
            res.next = l2 
            res = res.next 
            res.val += temp 
            temp = 0 
            if res.val >= 10 : 
                res.val -= 10 
                temp = 1 
            l2 = l2.next 
        if temp : 
            res.next = ListNode(temp) 
        return start 

File: C++/002_Add_Two_Numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *res = NULL , *tmp = new ListNode(0) ;
        res = tmp ; 
        int last = 0 ; 
        while ( l1 || l2 ){
            int key1 = ( l1 ) ? l1->val : 0 ; 
            int key2 = ( l2 ) ? l2->val : 0 ; 
            tmp->next = new ListNode( (key1+key2+last) % 10 ) ; 
            last = (key1+key2+last)/10 ; 
            tmp = tmp->next ; 
            if ( l1 ) l1 = l1->next ; 
            if ( l2 ) l2 = l2->next ; 
        } 
        if ( last > 0 )
            tmp->next = new ListNode(last) ; 
        return res->next ; 
    }
};